myhoard/restore_coordinator.py

```diff
@@ -1263,15 +1264,16 @@ def _check_sql_slave_status(self):
                 cursor.execute("STOP SLAVE")
                 # Current file could've been updated since we checked it the first time before slave was stopped.
                 # Get the latest value here so that we're sure to start from correct index
-                cursor.execute("SHOW SLAVE STATUS")
-                current_file = cursor.fetchone()["Relay_Log_File"]
-                last_index = int(current_file.rsplit(".", 1)[-1])
-                if last_index > current_index:
-                    self.log.info("Relay index incremented from %s to %s after STOP SLAVE", current_index, last_index)
-                    current_index = last_index
+                replica_status = get_slave_status(cursor)
+                if replica_status:
```

Comment:
The only case where this could be falsy without triggering the assert/exception above is if MySQL was restarted and forgot completely about its replication state, so something is odd and the state of the restoration is uncertain. I think we should fail here too and not just skip some part of the code.

Reply:
Done